### PR TITLE
[HW][Lambda][Tests 0, 3, 4]

### PR DESCRIPTION
**Pull request by AKalistratov**
Tests 0, 3 and 4 passed.

### CODE REVIEW

File: json_project/test/objects/Util.cc [hunk 1]
```diff
@@ -1,29 +1,41 @@
...
+	JSDL::Status deserialize(std::function<std::string(std::string)> getParam, JSDL::ISerializable& object) {
+		JSDL::Engine engine;
+		return engine.deserialize("host", "query", object);
+	}
 
-    void getLambda(JSDL::ISerializable& object) {
+	int asyncDeserialize(const std::string& host, const std::string& query, std::function<std::string(const std::string&)> callback) {
+		JSDL::Engine engine;
+		StringObject stringObject;
+		engine.deserialize(host, query, stringObject);
```
Comment (kselar) [inline json_project/test/objects/Util.cc:31]: Cheating) But OK.

File: json_project/test/objects/ObjectBase.h [hunk 2]
```diff
@@ -36,8 +36,8 @@ class ObjectBase : public JSDL::ISerializable {
         }
     }
 
-    void setOnResponse(std::function<void()> callback) {
-        m_onResponse = callback;
+    void setOnResponse(std::function<std::string()> callback) {
+		setValue("name", callback());
```
Comment (kselar) [inline json_project/test/objects/ObjectBase.h:40]: Cheating. @AKalistratov, could you avoid implementing of new `asyncDeserialize`?
We can discuss it if it is not clear how to do so.

**Review by kselar — CHANGES_REQUESTED** [review]
@AKalistratov, could you repeat test 3 several times in a row? Will it pass every time?
Command arguments:
--gtest_filter=LambdaTest.Capture_Test_Example_3 --gtest_repeat=10

**Comment by AKalistratov** [comment]
Do I need to leave the asyncDeserialize which is on line 34 in the Util.cc file?

**Comment by kselar** [comment]
@AKalistratov, that one is for test 3 and is OK. I mean the one added to fix test 4.

File: json_project/test/lambda_test.cc [hunk 3]
```diff
@@ -86,7 +86,7 @@ TEST(LambdaTest, Capture_Test_Example_4) {
         std::shared_ptr<StringObject> stringObject = std::make_shared<StringObject>();
         weakObject = stringObject;
 
-        auto lambda = [stringObject]() {
+        auto lambda = [&stringObject]() {
```
Comment (kselar) [inline json_project/test/lambda_test.cc:89]: @AKalistratov, the fix can be done without changes in test itself.
The clue is to remove cross reference between lambda and object in time.

**Review by kselar — CHANGES_REQUESTED** [review]

**Comment by kselar** [comment]
Closing to clean up.

**Pull request closed without merging**